### PR TITLE
project-happy-thoughts-api - Felicia Liabäck Löwstett

### PR DESCRIPTION
**Pull request by liaback-lowstett**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,30 +1,131 @@
...
+const port = process.env.PORT || 8080;
+const app = express();
+
+//schema - types, validators, error messages, default values
+const thoughtSchema = mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, "Ops, you forgot to write a message."],
+    minlength: [5, "The messages needs to be at least five characters.",],
+    maxlength: [140, "The messages needs to be less than 140 characters."],
```
Comment (KarolCislowski) [inline server.js:24]: Nicely configured field 👍🏻

File: server.js [hunk 2]
```diff
@@ -1,30 +1,131 @@
...
+app.use(express.json());
+
+// list enpoints
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+// GET request - show thoughts, sort, limit to 20 messages
+app.get('/thoughts', async (req,res) => {
+  const allThoughts = await Thought.find().sort( { createdAt: -1 } ).limit(20)
```
Comment (KarolCislowski) [inline server.js:49]: 👍🏻

File: server.js [hunk 3]
```diff
@@ -1,30 +1,131 @@
...
+  } catch (error) {
+    res.status(400).json(error);
+  }
+});
+
+// POST request - increas hearts to the db
+app.post("/thoughts/:id/likes", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const updatedThought = await Thought.findOneAndUpdate({ _id: id }, { $inc: { hearts: 1 }}, { new: true });
```
Comment (KarolCislowski) [inline server.js:67]: 💯

File: server.js [hunk 4]
```diff
@@ -1,30 +1,131 @@
...
+//     );
+//     if (updatedThought) {
+//       res.json(updatedThought);
+//     } else {
+//       res.status(404).json({ message: "Not found" });
+//     }
+//   } catch (error) {
+//     res.status(400).json({ message: "Invalied requeset", error });
+//   }
+// });
```
Comment (KarolCislowski) [inline server.js:126]: Remember to restrict access to these endpoints because not all users should have access to these functionalities 🤓

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
The code looks great but you need to provide a link to the frontend connected to it.

**Comment by liaback-lowstett** [comment]
Add netlify link

**Review by KarolCislowski — APPROVED** [review]
Great APP 🥇

**Pull request closed without merging**